ui/src/app/guards/workspace-guard.service.ts

```diff
@@ -3,24 +3,24 @@ import {
   ActivatedRouteSnapshot, CanActivate, CanActivateChild, Router,
   RouterStateSnapshot
 } from '@angular/router';
-import {serverConfigStore} from 'app/utils/navigation';
-import {WorkspacesService} from 'generated';
+import {currentWorkspaceStore, serverConfigStore} from 'app/utils/navigation';
+import {WorkspaceAccessLevel} from 'generated/fetch';
 import {Observable} from 'rxjs/Observable';
 
 @Injectable()
 export class WorkspaceGuard implements CanActivate, CanActivateChild {
   constructor(
-    private workspaceService: WorkspacesService,
     private router: Router) {}
 
   canActivate(route: ActivatedRouteSnapshot, state: RouterStateSnapshot): Observable<boolean> {
     if (serverConfigStore.getValue().enableResearchReviewPrompt && route.routeConfig.path === 'data' ||
         route.routeConfig.path === 'notebooks') {
-      this.workspaceService.getWorkspace(route.params.ns, route.params.wsid).subscribe(workspace => {
-        if (workspace.accessLevel === 'OWNER'
-            && workspace.workspace.researchPurpose.needsReviewPrompt) {
+      currentWorkspaceStore.subscribe(workspace => {
```

Comment:
I suggest using an async function with await here. It seems desirable to block this guard until you have evaluated the workspace. Right now this is a background process. That said, this might not be feasible since I'm not sure where the workspace load gets triggered during the rendering flow. Try it out - if not, the current way is probably OK.